Formula/lilypond.rb

```diff
@@ -0,0 +1,92 @@
+class Lilypond < Formula
+  desc "Music notation for everyone"
+  homepage "https://lilypond.org"
+  url "https://lilypond.org/download/sources/v2.22/lilypond-2.22.1.tar.gz"
+  sha256 "72ac2d54c310c3141c0b782d4e0bef9002d5519cf46632759b1f03ef6969cc30"
+  license "GPL-3.0-or-later"
+
+  depends_on "bdw-gc" => :build
+  depends_on "bison" => :build # Lilypond requires bison 2.4.1 or above
+  depends_on "fontforge" => :build
+  depends_on "gettext" => :build
+  depends_on "glib" => :build
+  depends_on "gmp" => :build
+  depends_on "imagemagick" => :build
+  depends_on "libtool" => :build
+  depends_on "libunistring" => :build
+  depends_on "perl" => :build
+  depends_on "pkg-config" => :build
+  depends_on "readline" => :build
+  depends_on "t1utils" => :build
+  depends_on "texi2html" => :build
+  depends_on "texinfo" => :build # lilypond requires texinfo 6.1 or above
+  depends_on "texlive" => :build
+
+  depends_on "fontconfig"
+  depends_on "freetype"
+  depends_on "ghostscript"
+  depends_on "pango"
+  depends_on "python@3.9"
+
+  uses_from_macos "flex" => :build
+  uses_from_macos "libffi" => :build
+
+  conflicts_with "guile",
+    because: "Lilypond installs `guile` version 1.8.8"
+
+  resource "guile@1" do
+    url "https://ftp.gnu.org/gnu/guile/guile-1.8.8.tar.gz"
+    sha256 "c3471fed2e72e5b04ad133bbaaf16369e8360283679bcf19800bc1b381024050"
+  end
```

Comment:
What was wrong with using `guile@2` as in the earlier version of your PR?

In any case, even if we absolutely needed `guile@1`, this should be installed into `libexec` as @nwhetsell suggests.

Reply:
Lilypond needs guile@1 to run and compile: 
https://lilypond.org/doc/v2.23/Documentation/contributor/requirements-for-running-lilypond

> Guile
> Use version 1.8.8. It is possible to build LilyPond with version 2.x of Guile and most features will work, but this is not officially supported (yet). In particular, performance and especially the startup time will be worse.

I'm not very familiar with use of `libexec`, but I will try to install guile on it, and make lilypond to find guile.

Reply:
Now guile@1 is in the `libexec` folder.

Reply:
Didn't you have an earlier working version of this formula that used `guile@2`?

Reply:
Yes, but lilypond doesn't officially support `guile@2`, although it probably works (https://lilypond.org/doc/v2.22/Documentation/contributor/requirements-for-running-lilypond). I don't know exactly which feature doesn't work, but I've noticed that lilypond works faster with `guile@1`.

Reply:
We should use `guile@2` unless we know that something's actually broken: https://docs.brew.sh/Acceptable-Formulae#stuff-that-requires-vendored-versions-of-homebrew-formulae

Reply:
Restored to guile@2. The new test cover guile and passed without errors.